ceph.spec.in

```diff
@@ -113,9 +113,10 @@
 
 %{!?_udevrulesdir: %global _udevrulesdir /lib/udev/rules.d}
 %{!?tmpfiles_create: %global tmpfiles_create systemd-tmpfiles --create}
-%{!?python3_pkgversion: %global python3_pkgversion 3}
-%{!?python3_version_nodots: %global python3_version_nodots 3}
-%{!?python3_version: %global python3_version 3}
+
+%{!?python3_version_major: %global python3_version_major 3}
+%{!?python3_version: %global python3_version 3.8}
```

Comment:
does RHEL8 and CentOS8 offer python3.8 ? IIUC, they only offer python 3.6.8. see https://centos.pkgs.org/8/centos-baseos-x86_64/python3-libs-3.6.8-37.el8.i686.rpm.html

Reply:
I'm not sure about RHEL, but [CentOS 8 has support via AppStream repo](https://pkgs.org/search/?q=python3.8).

Reply:
i assume that we need to enable the AppStream repo for using it? is all the dependencies available in app stream repo?

Reply:
@tchaikov I just tried this with the [oficial CentOS8 container image](https://hub.docker.com/_/centos):
```sh
> docker run --rm -ti centos:8 bash
# yum install -y python3.8
...
Installed:
  python38-3.8.6-3.module_el8.4.0+665+abc3a503.x86_64                                                         
  python38-libs-3.8.6-3.module_el8.4.0+665+abc3a503.x86_64                                                    
  python38-pip-19.3.1-1.module_el8.4.0+647+0ba99ce8.noarch                                                    
  python38-pip-wheel-19.3.1-1.module_el8.4.0+647+0ba99ce8.noarch                                              
  python38-setuptools-41.6.0-4.module_el8.4.0+647+0ba99ce8.noarch                                             
  python38-setuptools-wheel-41.6.0-4.module_el8.4.0+647+0ba99ce8.noarch                                       

Complete!
```
So it seems that the AppStream repo is enabled by default.

Reply:
i see. just read more about AppStream repo. yeah, it seems that it is enabled by default. but i am not sure what the "quality of service" we can expect from it.

1. are all python packages available we need in BaseOS also available in AppStream?
2. what about their lifecycles?
3. do they also get the backport and fixes?

Reply:
> i see. just read more about AppStream repo. yeah, it seems that it is enabled by default. but i am not sure what the "quality of service" we can expect from it.

`RHEL 8 content is distributed through the two main repositories: BaseOS and AppStream.` [[ref]](https://docs.centos.org/en-US/8-docs/managing-userspace-components/assembly_using-appstream/)

> 1. are all python packages available we need in BaseOS also available in AppStream?

It has even more: BaseOS has 87 python3-* packages, while AppStream has 267.

> 2. what about their lifecycles?
> 3. do they also get the backport and fixes?

> "New versions of most application streams will become available as needed during the full 10 years of the Red Hat Enterprise Linux 8 life. Lifecycles will align to the upstream support life or to product needs as required. Multiple versions of the same application stream may be available at the same time with overlapping lives. Rolling streams for fast-evolving products have only a single version, the newest, available. If a non-security related bug is discovered in a rolling stream, we will work to address it in the next scheduled release. The release cadence will be selected based on what makes sense for the product. Rolling streams will be available and supported for the full life of Red Hat Enterprise Linux major releases."

 [[ref]](https://access.redhat.com/support/policy/updates/rhel8-app-streams-life-cycle)